### PR TITLE
api: fill up API documentation

### PR DESCRIPTION
**Pull request by eliben**
Followup for #2878

Now that the documentation is more complete, mention it in the README. Once a new version of ollama is tagged, pkg.go.dev will pick up the documentation comments and display everything on the linked page

Updates #2840

### CODE REVIEW

**Comment by eliben** [comment]
@jmorganca @pdevine friendly ping for a review

**Comment by eliben** [comment]
Conflict vs. main resolved, now the PR is up to date

File: README.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ The official [Ollama Docker image](https://hub.docker.com/r/ollama/ollama) `olla
 
 - [ollama-python](https://github.com/ollama/ollama-python)
 - [ollama-js](https://github.com/ollama/ollama-js)
+- [Go client API](https://pkg.go.dev/github.com/ollama/ollama/api)
```
Comment (jmorganca) [inline README.md:35]:
```suggestion
```

Ok if we list this in a follow up?

**Review by jmorganca — APPROVED** [review]

**Comment by jmorganca** [comment]
Thanks @eliben !

**Pull request merged**